Content.Server/MachineLinking/System/SignalLinkerSystem.cs

```diff
@@ -20,262 +20,176 @@ namespace Content.Server.MachineLinking.System
+            foreach (var receiverPort in component.Inputs)
+                foreach (var transmitterPort in receiverPort.Transmitters)
+                    transmitterPort.Receivers.Remove((uid, receiverPort.Name));
         }
 
-        private void OnTransmitterInvokePort(EntityUid uid, SignalTransmitterComponent component, InvokePortEvent args)
+        private void OnTransmitterInteractUsing(EntityUid uid, SignalTransmitterComponent component, InteractUsingEvent args)
         {
-            if (!component.Outputs.TryGetPort(args.Port, out var port)) throw new PortNotFoundException();
+            if (args.Handled) return;
 
-            if (args.Value == null)
-            {
-                if (port.Type != null && !port.Type.IsNullable()) throw new InvalidPortValueException();
-            }
-            else
+            if (!TryComp(args.Used, out SignalLinkerComponent? linker) ||
+                !TryComp(args.User, out ActorComponent? actor))
+                return;
+
+            if (component.Outputs.Count == 1)
             {
-                if (port.Type == null || !args.Value.GetType().IsAssignableTo(port.Type))
-                    throw new InvalidPortValueException();
+                var port = component.Outputs.First();
+                LinkerTransmitterInteraction(args.User, linker, component, port.Name);
+                args.Handled = true;
+                return;
             }
 
-            port.Signal = args.Value;
-
-            foreach (var link in _linkCollection.GetLinks(component, port.Name))
+            if (_userInterfaceSystem.TryGetUi(uid, SignalTransmitterUiKey.Key, out var bui))
             {
-                if (!IsInRange(component, link.ReceiverComponent)) continue;
-
-                RaiseLocalEvent(link.ReceiverComponent.Owner,
-                    new SignalReceivedEvent(link.Receiverport.Name, args.Value), false);
+                bui.Open(actor.PlayerSession);
+                bui.SetState(new SignalPortsState(component.Outputs.Select(port => port.Name).ToArray()));
+                args.Handled = true;
+                return;
             }
         }
 
         private void OnReceiverInteractUsing(EntityUid uid, SignalReceiverComponent component, InteractUsingEvent args)
         {
             if (args.Handled) return;
 
-            if (!EntityManager.TryGetComponent<SignalLinkerComponent?>(args.Used, out var linker) || !linker.Port.HasValue ||
-                !EntityManager.TryGetComponent(args.User, out ActorComponent? actor) ||
-                !linker.Port.Value.transmitter.Outputs.TryGetPort(linker.Port.Value.port, out var port))
-            {
+            if (!TryComp(args.Used, out SignalLinkerComponent? linker) || !linker.savedPort.HasValue ||
+                !TryComp(args.User, out ActorComponent? actor) ||
+                !linker.savedPort.Value.transmitter.Outputs.TryGetPort(linker.savedPort.Value.port, out var port))
                 return;
```

Comment:
Because the linker/multitool could be using an comp from an old/deleted entity, this needs to check `linker.Port.Value.Transmitter.Deleted`.

Reply:
Or alternatively, store the entityUid & port name, instead of component & port name